cpp/src/arrow/array/array_test.cc

```diff
@@ -410,6 +410,7 @@ static std::vector<std::shared_ptr<DataType>> TestArrayUtilitiesAgainstTheseType
 
 TEST_F(TestArray, TestMakeArrayOfNull) {
   for (int64_t length : {0, 1, 16, 133}) {
+    ARROW_SCOPED_TRACE("length = ", length);
     for (auto type : TestArrayUtilitiesAgainstTheseTypes()) {
```

Comment:
It seems `TestArrayUtilitiesAgainstTheseTypes` doesn't include an empty struct, can we add one there? (or if it fails other tests, test `MakeArrayOfNull` specifically with an empty struct)